contrib/storage-splunk/src/main/java/org/apache/drill/exec/store/splunk/SplunkSchema.java

```diff
@@ -148,8 +183,24 @@ private void registerIndexes() {
     registerTable(SPL_TABLE_NAME, new DynamicDrillTable(plugin, plugin.getName(),
       new SplunkScanSpec(plugin.getName(), SPL_TABLE_NAME, plugin.getConfig(), queryUserName)));
 
+    Set<String> indexList = null;
     // Retrieve and add all other Splunk indexes
-    for (String indexName : connection.getIndexes().keySet()) {
+    // First check the cache to see if we have a list of indexes.
+    String nameKey = getNameForCache();
+    if (useCache) {
+      indexList = cache.getIfPresent(nameKey);
+    }
```

Comment:
@cgivre is the cache really storing one Splunk index per key? Here it looks to me like there's a single cache key derived from the queryUserName and the plugin name that holds a list of indexes. Or am I just being confused by the Caffeine API?

Reply:
@jnturton , 
I'm sorry, I misspoke.   The way I intended the cache to work was that we combine the plugin name + user name to create a key, and the value is a list of indexes.   Every time a user adds or drops an index, we have to recreate the cache entry for that plugin/username.

So:
```
splunk1-cgivre: [index1, index2, index2]
splunk2-cgivre: [index5, index6, index7]
splunk1-jnturton: [index1, index2]
```

That's what the cache should look like.  (In theory)

Reply:
Okay, in that case, can I propose 

- a Map of caches with one cache per plugin name + query user and one cache key per Splunk index or
- a unified cache with one cache key per plugin name + query user + Splunk index name?

Reply:
@jnturton I'm remembering why I did it this way.  When the plugin is accessed, Drill calls the `registerTable` method which loads the list of indexes in to memory.   From my recollection, all similar storage plugins, do something similar in that they load the schemata into memory when the plugin is first accessed.   Here the change I made was to first check the memory cache and if that's not populated, then call the index list from Splunk.  

I'm happy to do this as you suggest but I'm a little confused as to what exactly you're asking for.

Reply:
Let's go with this implementation. Fine tuning can come later if it's needed.